DockerHubBackend/DockerHubBackend/Repository/Implementation/DockerRepositoryRepository.cs

```diff
@@ -21,5 +21,68 @@ public DockerRepository GetFullDockerRepositoryById(Guid id)
                 .Include(dockerRepository => dockerRepository.Images)
                 .FirstOrDefault(dockerRepository => dockerRepository.Id == id);
         }
+
+        public List<DockerRepository> GetStarRepositoriesForUser(Guid userId)
+        {
+            return _context.Users
+                .OfType<StandardUser>()
+                .Where(user => !user.IsDeleted)
+                .AsQueryable()
+                .Include(user => user.StarredRepositories)
+                    .ThenInclude(starRepository => starRepository.UserOwner)
+                .Include(user => user.StarredRepositories)
+                    .ThenInclude(starRepository => starRepository.OrganizationOwner)
+                .Include(user => user.StarredRepositories)
+                    .ThenInclude(starRepository => starRepository.Images)
+                .First(user => user.Id == userId)
+                .StarredRepositories
+                .ToList();
+        }
+
+        public List<DockerRepository> GetPrivateRepositoriesForUser(Guid userId)
+        {
+            return _context.DockerRepositories
+                .AsQueryable()
+                .Include(dockerRepository => dockerRepository.UserOwner)
+                .Include(dockerRepository => dockerRepository.OrganizationOwner)
+                    .ThenInclude(organization => organization.Owner)
+                .Include(dockerRepository => dockerRepository.Images)
+                .Where(dockerRepository => !dockerRepository.IsDeleted)
+                .Where(dockerRepository => dockerRepository.UserOwnerId == userId ||
+                                           dockerRepository.OrganizationOwner.OwnerId == userId)
```

Comment:
Will this throw a null reference error if OrganizationOwner is null?

Reply:
It should have been `dockerRepository.OrganizationOwnerId == userId`, although, I don't think this will throw null reference error because if first condition is false then the second one should be true (if database is populated correctly).